dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java

```diff
@@ -1458,4 +1616,25 @@ private long countMaySet() {
     private long countMustSet() {
         return knowledgeMap.values().stream().mapToLong(k -> k.must.size()).sum();
     }
+
+    private Boolean directAlias(MemEvent e1, MemEvent e2) {
```

Comment:
is this the same as `AliasAnalysis.virtualLoc`?

Reply:
AliasAnalysis.virtualLoc is `same_alias_r` of Alloy, illustrating relation between generic location and the location on  special-purpose caching (other proxy) alias to it.
The directAlias here is `same_location_r` of Alloy, illustrating relation between two generic locations; or if one or both of them are non-generic proxy location, illustrating the relation of generic locations they alignling to.